lib/sdp/media.ex

```diff
@@ -159,4 +168,68 @@ defmodule Membrane.Protocol.SDP.Media do
   end
 
   defp gen_ports(port_no, _), do: [port_no]
+
+  defp header_fields(media) do
+    [
+      serialize_type(media.type),
+      serialize_ports(media.ports),
+      media.protocol,
+      serialize_fmt(media.fmt)
+    ]
+  end
+
+  defp other_fields(media) do
+    [
+      {"i", :title, nil},
+      {"c", :connection_data, Membrane.Protocol.SDP.ConnectionData},
+      {"b", :bandwidth, Membrane.Protocol.SDP.Bandwidth},
+      {"k", :encryption, Membrane.Protocol.SDP.Encryption},
+      {"a", :attributes, Membrane.Protocol.SDP.Attribute}
+    ]
+    |> Enum.map(&add_types(&1, media))
+    |> Enum.reject(&(&1 == ""))
+  end
+
+  def add_types({type_string, key, module}, media) do
+    case Map.get(media, key) do
+      nil ->
+        ""
+
+      list when is_list(list) ->
+        list
+        |> Enum.map(&serialize_optional(&1, module))
+        |> Enum.reject(&(&1 == ""))
+        |> Enum.map_join(&add_type(&1, type_string))
+
+      single_field ->
+        single_field |> serialize_optional(module) |> add_type(type_string)
+    end
+  end
+
+  defp add_type(string, type) do
+    if String.at(string, 1) == "=" do
+      string
+    else
+      type <> "=" <> string
+    end
+  end
+
+  defp serialize_optional(value, nil), do: to_string(value)
+  defp serialize_optional(value, module), do: apply(module, :serialize, [value])
+
+  defp serialize_type(type) when is_binary(type), do: type
+  defp serialize_type(type) when is_atom(type), do: Atom.to_string(type)
+
+  defp serialize_ports(ports) do
+    port = ports |> Enum.sort() |> hd |> Integer.to_string()
+
+    if length(ports) == 1 do
```

Comment:
If you used pattern matching on first element, you could avoid using `hd/1` and at the same time this if with `length/1` call would change into prettier `case`.